experimental/LieAlgebras/src/root_systems.jl

```diff
@@ -0,0 +1,224 @@
+mutable struct RootSystem
+  roots::Vector{Vector{Int}}
+  simple_roots::Vector{Vector{Int}}
+  positive_roots::Vector{Vector{Int}}
+  root_system_type::Tuple{Symbol,Int64}
+  GAP_root_system::GAP.GapObj
+  function RootSystem(S::Symbol, n::Int64)
+    # S is a symbol detailing the type of the indecomposable root system 
+    # e.g. "A", "B", "C",... and n is an integer for the number of simple roots
+    S1 = GAP.Obj(S)
+    RS = GAP.Globals.RootSystem(S1, n)
+    sR = Vector{Vector{Int}}(GAP.Globals.SimpleSystem(RS))
+    Ro1 = Vector{Vector{Int}}(GAP.Globals.PositiveRoots(RS))
+    Ro2 = Vector{Vector{Int}}(GAP.Globals.NegativeRoots(RS))
+    Ro = vcat(Ro1, Ro2)
+    t = (S, n)
+    return new(Ro, sR, Ro1, t, RS)
+  end
+end
+
+###############################################################################
+#
+#   Basic manipulation
+#
+###############################################################################
+@doc raw"""
+    number_of_roots(R::RootSystem)
+
+Return the numbers of roots in the root system `R`.
+"""
+number_of_roots(R::RootSystem) = size(R.roots)[1]
+
+@doc raw"""
+    number_of_roots(S::Symbol, n::Int64)
+
+Return the numbers of roots in the root system of type `S`
+"""
+number_of_roots(S::Symbol, n::Int64) = number_of_roots(RootSystem(S, n))
```

Comment:
```suggestion
number_of_roots(S::Symbol, n::Int) = number_of_roots(root_system(S, n))
```
the snake_case version does input sanitation